src/main/java/com/treasure/hunt/game/GameManager.java

```diff
@@ -239,9 +220,37 @@ public BooleanBinding stepBackwardImpossibleBinding() {
     }
 
     /**
-     * @return true if the shown step is the first one
+     * @return {@code true}, if the shown step is the first one. {@code false}, otherwise.
      */
     public boolean isFirstStepShown() {
         return stepBackwardImpossibleBinding().getValue();
     }
+
+    /**
+     * @param coordinate the point on the canvas, we want to get the closest {@link GeometryType} to.
+     * @param distance   the maximum distance to a potential {@link GeometryItem}.
+     * @return the nearest {@link GeometryItem} to {@code (x,y)}, with a maximum distance of {@code distance}.
+     */
+    public GeometryItem pickGeometryItem(Coordinate coordinate, double distance) {
+        if (moves.size() < 1) {
+            return null;
+        }
+
+        Point mouse = JTSUtils.GEOMETRY_FACTORY.createPoint(coordinate);
+
+        GeometryItem nearestGeometryItem = moves.get(0).getGeometryItems().get(0);
+        for (Move move : moves.subList(0, viewIndex.get() + 1)) {
+            for (GeometryItem geometryItem : move.getGeometryItems()) {
+                if (mouse.distance(geometryItem.getGeometry()) < mouse.distance(nearestGeometryItem.getGeometry())
+                    /*&& geometryItem.getGeometryStyle().isVisible()*/) {
```

Comment:
No commented out code we have version Control

Reply:
oops, this should not be commented out, thanks